devito/ops/node_factory.py

```diff
@@ -0,0 +1,214 @@
+            Number to initialize an Float node.
+
+        Returns
+        -------
+        :class:`Float`
+            Object of :class:`Float`.
+        """
+        return Float(number)
+
+    def new_rational_node(self, num, den):
+        """
+        Creates a new sympy :class:`Rational` object to represent 
+        a mathematical rational.
+
+        Parameters
+        ----------
+        num : float
+            Rational numerator.
+        den : float
+            Rational denominator.
+
+        Returns
+        -------
+        :class:`Rational`
+            Object :class:`Rational`. 
+        """
+        return Rational(num, den)
+
+    def new_add_node(self, lhs, rhs):
+        """
+        Represents a new mathematical sum node operation.
+
+        Parameters
+        ----------
+        lhs 
+            The left hand side of the sum operation.
+        rhs : [type]
+            The right hand side of the sum operation.
+
+        Returns
+        -------
+        :class:`Add`
+            Object of :class:`Add`.
+        """
+        return Add(lhs, rhs)
+
+    def new_mul_node(self, lhs, rhs):
+        """
+        Creates a new mathematical multiplication node operation.
+
+        Parameters
+        ----------
+        lhs
+            The left hand side of the multiplication.
+        rhs : [type]
+            The right hand side of the multiplication.
+
+        Returns
+        -------
+        :class:`Mul`
+            Object of :class:`Mul`.
+        """
+        return Mul(lhs, rhs)
+
+    def new_divide_node(self, num, den):
+        """
+        Creates a new division node operation.
+
+        Notes
+        -----
+        The division is represented by the multiplicating the numerator with 
+        denominator's inverse.
+
+        Parameters
+        ----------
+        num
+            The division numerator.
+        den : [type]
+            The division denominator.
+
+        Returns
+        -------
+        :class:`Mul`
+            Object of :class:`Mul` representing the division.
+        """
+        return Mul(num, Pow(den, Integer(-1)))
+
+    def new_grid(self, devito_indexed, dimensions):
+        """
+        Creates an :class:`Indexed` object using OPS representation.
+        If the pair grid name and time dimension was alredy created, it will return
+        the stored value associated with this pair.
+
+        Parameters
+        ----------
+        devito_indexed : :class:`Indexed`
+            Indexed object using devito representation.
+
+        dimensions : list
+            List of the dimensions alredy translated to OPS.
+
+        Returns
+        -------
+        :class:`Indexed`
+            Indexed object using OPS representation.
+        """
+
+        def getDimensionsDisplacement(dimensions):
+            disp = []
+
+            for dim in dimensions:
+                if dim.is_Add:
+                    lhs, = dim.as_two_terms()
+                    disp.append(str(lhs))
```

Comment:
I'm not sure this works. What if the Add has multiple addends, such as ``a + b + 2 + c + d[4] + ...`` ?